vm/package/deploy.rs

```diff
@@ -113,19 +128,25 @@ impl<'de, N: Network> Deserialize<'de> for DeployResponse<N> {
     fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
```

Comment:
We should also check that the deserialized transaction here is a `Transaction::Deploy`